api/Dockerfile.dev

```diff
@@ -0,0 +1,13 @@
+ARG RUST_VERSION=1.78.0
+
+FROM rust:${RUST_VERSION}-slim-bookworm
+
+WORKDIR /app
+
+COPY . .
```

Comment:
Prefer binding at compose.dev.yaml level instead of copying the workspace in the development image.

Reply:
> Prefer binding at compose.dev.yaml level instead of copying the workspace in the development image.

The watch feature in Docker Compose is designed to avoid volume binding due to its performance issues (source: [Docker Blog](https://www.docker.com/blog/announcing-docker-compose-watch-ga-release/)). I've taken the example of the compose documentation page : https://docs.docker.com/compose/file-watch/ 

> Also why cargo install --path . since it's a ""devcontainer"" ?

Even in a development container, we need to install the cargo dependencies. Running `cargo install` results in an error that can be resolved by using `cargo install --path .`
However, as I'm not a Rust expert, there might be a more optimal approach.

Reply:
Fait enough, I've poor windows dev experience I did not see the WSL problem. Thanks for documenting it !

Reply:
> Even in a development container, we need to install the cargo dependencies. 

Well `cargo install` is for installing the built executable somewhere, it is just doing `cargo build` + copying artefacts.

Same as `cargo run`, it will `cargo build` before running the executable if it was not previously built.
A simple `cargo build` should suffice, and I'm not even sure why you need to build it before `cargo watch`

Reply:
Maybe you're right on this ! Will test on my side